Windows: Fix output values reported by PcpPrimIndex::PrintStatistics()

HamedSabri-adsk opened the pull request:
### Description of Change(s)

This PR fixes the values displayed incorrectly when PcpPrimIndex::PrintStatistics() routine is called on Windows. 

According to Linux manual page for printf 

```
      '      For decimal conversion (i, d, u, f, F, g, G) the output is
              to be grouped with thousands' grouping characters if the
              locale information indicates any.  (See setlocale(3).)
              Note that many versions of gcc(1) cannot parse this option
              and will issue a warning.  (SUSv2 did not include %'F, but
              SUSv3 added it.)
```

https://man7.org/linux/man-pages/man3/printf.3.html

There is no such flag for Windows:

https://docs.microsoft.com/en-us/cpp/c-runtime-library/format-specification-syntax-printf-and-wprintf-functions?view=msvc-160

hence the output values show up wrong on Windows and correct on Linux:

**Windows:**
![windows_statistic_printf](https://user-images.githubusercontent.com/48299423/112030562-c70a2300-8b10-11eb-82b7-34f94bcc78dc.PNG)

**Linux:**
![linux_statistic_printf](https://user-images.githubusercontent.com/48299423/112032930-47ca1e80-8b13-11eb-92cf-3e99228ad17e.png)

**After the Fix:**

![fix_me](https://user-images.githubusercontent.com/48299423/112033064-6af4ce00-8b13-11eb-8871-7920759d7ab7.PNG)

Commit:
Windows: Fix output values reported by PcpPrimIndex::PrintStatistics()

HamedSabri-adsk commented:
Hi @spiffmon, here is the fix that I mentioned earlier in our chat on google interest group yesterday. 

Cheers,
H

jtran56 commented:
Filed as internal issue #USD-6617

c64kernal commented:
Thanks for the fix @HamedSabri-adsk! We generally try very hard to avoid any platform-specific ifdefs outside of arch. Would it be possible to just remove offending "''" and instead do something a little more standard across all platforms but still visually easy to compare (e.g., everything right justified)? Thanks again!

Commit:
Remove #ifdef and simplify the code by removing the offending "'".

HamedSabri-adsk commented:
Hi @c64kernal, 

sure, that's a good point.  I removed the offending "'"  all together. As for the text Alignment, I didn't want to make a bigger change than it is necessary since the alignment looks visually nice and readable even with a very large number (e.g 15 digits). Hope this Ok?

Addressed: 76705d8198959d0f3918256487bf65d9945ee1b9

```
  Total nodes:                       922337200000000
  Total culled* nodes:               922337200000000
  By type (total / culled*):         
    root:                            922337200000000 / 922337200000000
    inherit:                         922337200000000 / 922337200000000
      implied inherits:              922337200000000 / 922337200000000
    variant:                         922337200000000 / 922337200000000
    relocate:                        922337200000000 / 922337200000000
    reference:                       922337200000000 / 922337200000000
    payload:                         922337200000000 / 922337200000000
    specialize:                      922337200000000 / 922337200000000
  (*) This does not include culled nodes that were erased from the graph
```

c64kernal commented:
This seems totally reasonable to me, thanks so much @HamedSabri-adsk! We'll try to get it merged soon.
